CRM/Contribute/Form/Search.php

```diff
@@ -390,6 +390,11 @@ public function fixFormValues() {
       $this->_defaults['contribution_status_id'] = array($status => 1);
     }
 
+    $pcpid=CRM_Utils_Request::retrieve('pcpid', 'Positive', $this);
+    if ($pcpid) {
```

Comment:
Please, check the source file:
https://github.com/civicrm/civicrm-core/blob/master/CRM/Utils/Type.php

To confirm types available, and check the source and you will see that contribution_pcp_made_through_id field[] is an array, not simple input:

Current form template:
`
<select class="crm-select2 crm-form-multiselect" multiple="multiple" placeholder="- any -" name="contribution_pcp_made_through_id[]" id="contribution_pcp_made_through_id" tabindex="-1" style="display: none;">
	<option value="1">Campaign name</option>
</select>
` 
Regards